File: app/src/main/java/fudi/freddy/mokitosample/store/SQLiteContact.java
package fudi.freddy.mokitosample.store;

import fudi.freddy.mokitosample.model.Contact;

/**
 * @version 1.0
 * @autor Romell Domínguez
 * @date 2/26/17
 */

public class SQLiteContact implements StoreContract<Contact> {
    @Override
    public void save(Contact object) {
        DatabaseHelper.getInstance().createContact(object);
    }

    @Override
    public Contact load(String mail) {
        return DatabaseHelper.getInstance().getContact(mail);
    }
}

File: app-kotlin/src/main/java/fudi/freddy/mokitosample/store/SQLiteContact.kt
package fudi.freddy.mokitosample.store

import fudi.freddy.mokitosample.model.Contact

/**
 * @version 1.0
 * @autor Romell Domínguez
 * @date 7/25/20
 */
class SQLiteContact : StoreContract<Contact?> {
    override fun save(`object`: Contact?) {
        DatabaseHelper.instance!!.createContact(`object`)
    }

    override fun load(mail: String): Contact? {
        return DatabaseHelper.instance!!.getContact(mail)
    }
}